pkg/instancetype/instancetype_test.go

```diff
@@ -280,6 +285,41 @@ var _ = Describe("Instancetype and Preferences", func() {
 				}
 				Expect(instancetypeMethods.StoreControllerRevisions(vm)).To(MatchError(ContainSubstring("VM field conflicts with selected Instancetype")))
 			})
+
+			It("find successfully decodes v1alpha1 VirtualMachineInstancetypeSpecRevision ControllerRevision without APIVersion set - bug #9261", func() {
+				specData, err := json.Marshal(clusterInstancetype.Spec)
+				Expect(err).ToNot(HaveOccurred())
+
+				specRevision := instancetypev1alpha1.VirtualMachineInstancetypeSpecRevision{
+					Spec: specData,
+				}
+				specRevisionData, err := json.Marshal(specRevision)
+				Expect(err).ToNot(HaveOccurred())
+
+				controllerRevision := &appsv1.ControllerRevision{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:            "crName",
+						Namespace:       vm.Namespace,
+						OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(vm, v1.VirtualMachineGroupVersionKind)},
+					},
+					Data: runtime.RawExtension{
+						Raw: specRevisionData,
+					},
+				}
+
+				_, err = virtClient.AppsV1().ControllerRevisions(vm.Namespace).Create(context.Background(), controllerRevision, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				vm.Spec.Instancetype = &v1.InstancetypeMatcher{
+					Name:         clusterInstancetype.Name,
+					RevisionName: controllerRevision.Name,
+					Kind:         apiinstancetype.ClusterSingularResourceName,
+				}
+
+				foundInstancetypeSpec, err := instancetypeMethods.FindInstancetypeSpec(vm)
```

Comment:
Can you somehow assert that the `apiVersion` is empty?

Reply:
L293 leaves it empty within the `VirtualMachineInstancetypeSpecRevision` stashed within the `ControllerRevision`. I can't think why we'd need to assert that again here?

Reply:
IMO explicitly stating something is cleaner than implicitly relying on it. Yes it is left empty in L293 but it is not obvious at first glance. At least for me.

Reply:
I agree but it's not a property of `foundInstancetypeSpec` that we can assert, to assert that the field isn't set on the stashed `VirtualMachineInstancetypeSpecRevision` within the `ControllerRevision` that we've built above just seems redundant in this context.

Reply:
Ok, then maybe add a comment to at least indicate that the apiVersion is left out on purpose?